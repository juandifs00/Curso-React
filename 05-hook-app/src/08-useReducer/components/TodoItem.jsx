const TodoItem = ({ state, onDeleteTodo, onToggleTodo }) => {
  return (
    <li className="list-group-item d-flex justify-content-between">
      <span
        className={`align-self-center ${
          state.done ? "text-decoration-line-through" : ""
        }`}
        onDoubleClick={() => onToggleTodo(state.id)}
        aria-label="span"
      >
        {state.description}
      </span>
      <button onClick={() => onDeleteTodo(state.id)} className="btn btn-danger">
        Borrar
      </button>
    </li>
  );
};

export default TodoItem;
