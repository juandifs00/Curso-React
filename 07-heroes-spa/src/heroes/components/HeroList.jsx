import HeroCard from "./HeroCard";
import { getHeroesByPublisher } from "../helpers/getHeroesByPublisher";

const HeroList = ({ publisher }) => {
  if (!publisher) return <></>;
  const heroes = getHeroesByPublisher(publisher);

  return (
    <div className="row row-cols-1 row-cols-md-3 g-3">
      {heroes.map((hero) => (
        <HeroCard key={hero.id} {...hero} />
      ))}
    </div>
  );
};

export default HeroList;
