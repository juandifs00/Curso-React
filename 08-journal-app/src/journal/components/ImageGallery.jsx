import ImageList from "@mui/material/ImageList";
import ImageListItem from "@mui/material/ImageListItem";

const ImageGallery = ({ images }) => {
  return (
    <ImageList sx={{ width: '100%', height: 500 }} cols={4} rowHeight={360}>
      {images.map((image) => (
        <ImageListItem key={image}>
          <img
            srcSet={`${image}?w=164&h=164&fit=crop&auto=format&dpr=2 2x`}
            src={`${image}?w=164&h=164&fit=crop&auto=format`}
            alt="Imagen de la nota"
            loading="lazy"
          />
        </ImageListItem>
      ))}
    </ImageList>
  );
};

export default ImageGallery;
