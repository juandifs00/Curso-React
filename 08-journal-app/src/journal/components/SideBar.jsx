import { useSelector } from "react-redux";
import { TurnedInNot } from "@mui/icons-material";
import {
  Box,
  Divider,
  Drawer,
  Grid,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  Toolbar,
  Typography,
} from "@mui/material";

const SideBar = ({ drawerWidth = 240 }) => {
  const { displayName } = useSelector((state) => state.auth);

  return (
    <Box
      component="nav"
      sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
    >
      <Drawer
        variant="persistent"
        open
        sx={{
          display: { xs: "block" },
          "& .MuiDrawer-paper": { width: drawerWidth, boxSizing: "border-box" },
        }}
      >
        <Toolbar>
          <Typography variant="h6" noWrap component="div">
            {displayName}
          </Typography>
        </Toolbar>
        <Divider />

        <List>
          {[
            "Enero",
            "Febrero",
            "Marzo",
            "Abril",
            "Mayo",
            "Junio",
            "Julio",
            "Agosto",
            "Septiembre",
            "Octubre",
            "Noviembre",
            "Diciembre",
          ].map((text) => (
            <ListItem key={text} disablePadding>
              <ListItemButton>
                <ListItemIcon>
                  <TurnedInNot />
                </ListItemIcon>
                <Grid container>
                  <ListItemText primary={text} />
                  <ListItemText
                    secondary={
                      "Commodo sint mollit proident ipsum adipisicing do magna esse officia cupidatat."
                    }
                  />
                </Grid>
              </ListItemButton>
            </ListItem>
          ))}
        </List>
      </Drawer>
    </Box>
  );
};

export default SideBar;
