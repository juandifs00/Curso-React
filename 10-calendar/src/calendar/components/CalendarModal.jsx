import { useState } from "react";
import Modal from "react-modal";

const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    marginRight: "-50%",
    transform: "translate(-50%, -50%)",
  },
};

Modal.setAppElement("#root");

export const CalendarModal = () => {
  const [isModalOpen, setIsModalOpen] = useState(true);

  const onCloseModal = () => {
    console.log("Modal cerrado");
    setIsModalOpen(false);
  };

  return (
    <Modal
      isOpen={isModalOpen}
      onRequestClose={onCloseModal}
      style={customStyles}
      className="modal"
      overlayClassName="modal-fondo"
      closeTimeoutMS={200} // Tiempo de cierre en milisegundos
    >
      <h1>Hola Mundo</h1>
      <hr />
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Nobis optio
        eligendi in. Excepturi?
      </p>
    </Modal>
  );
};
