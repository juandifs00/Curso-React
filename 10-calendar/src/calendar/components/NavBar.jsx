export const NavBar = () => {
  return (
    <div className="navbar navbar-dark bg-dark mb-4 px-4">
      <span className="navbar-brand text-white">
        <i className="fas fa-calendar-alt"></i>
        &nbsp; Calendar App &nbsp; Juandi
      </span>
      <button
        className="btn btn-outline-danger"
        onClick={() => console.log("Logout")}
      >
        <i className="fas fa-regular fa-arrow-right-from-bracket"></i>
        <span>Salir</span>
      </button>
    </div>
  );
};
