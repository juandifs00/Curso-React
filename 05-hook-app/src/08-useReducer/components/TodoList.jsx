import TodoItem from "./TodoItem";

const TodoList = ({ state = [], onDeleteTodo, onToggleTodo }) => {
  return (
    <ul className="list-group">
      {state.map((todo) => (
        // TodoItem
        <TodoItem
          key={state.id}
          state={todo}
          onDeleteTodo={onDeleteTodo}
          onToggleTodo={onToggleTodo}
        />
      ))}
    </ul>
  );
};

export default TodoList;
