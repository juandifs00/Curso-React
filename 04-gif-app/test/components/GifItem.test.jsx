import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en el componente <GifItem />", () => {
  const title = "XDXDXD";
  const url = "https://fast&furious.com/XDXDXD.jpg";

  test("Debe de hacer match con el SnapShot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
});
