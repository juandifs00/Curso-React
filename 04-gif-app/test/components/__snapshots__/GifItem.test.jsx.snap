// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem /> Debe de hacer match con el SnapShot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="XDXDXD"
      src="https://fast&furious.com/XDXDXD.jpg"
    />
    <p>
      XDXDXD
    </p>
  </div>
</div>
`;
